core/templates/components/summary-tile/subtopic-summary-tile.component.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a subtopic tile.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { TopicViewerDomainConstants } from 'domain/topic_viewer/topic-viewer-domain.constants';
+import { AssetsBackendApiService } from 'services/assets-backend-api.service';
+import { AppConstants } from 'app.constants';
+import { Subtopic } from 'domain/topic/subtopic.model';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'oppia-subtopic-summary-tile',
+  templateUrl: './subtopic-summary-tile.component.html'
+})
+export class SubtopicSummaryTileComponent implements OnInit {
+  @Input() classroomUrlFragment!: string;
+  @Input() subtopic!: Subtopic;
+  @Input() topicId!: string;
+  @Input() topicUrlFragment!: string;
+  thumbnailUrl!: string;
+  thumbnailBgColor!: string;
+  subtopicTitle !: string;
```

Comment:
Do we need `!`?

Reply:
Yup, the properties are undefined therefore we need to do non-null assertion